codegen/src/codegen_rust.rs

```diff
@@ -1218,7 +1218,7 @@ impl<'model> RustCodeGen<'model> {
         writeln!(
             w,
             r#"/// client for sending {} messages
-              #[derive(Debug)]
+              #[derive(Clone,Debug)]
```

Comment:
I think `T` needs to be `Clone` for this to work

Reply:
Great catch. This needs a bit more work because the provider impl of Transport isn't Clone.